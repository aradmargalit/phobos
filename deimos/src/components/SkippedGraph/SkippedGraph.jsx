import { meanBy as _meanBy, startCase as _startCase } from 'lodash';
import React from 'react';

import IntervalGraph from '../IntervalGraph';

const average = data => _meanBy(data, 'percentage');

export default function SkippedGraph({ loading, intervalData, intervalType }) {
  const data = intervalData.map(({ interval, percentage_active: percentageActive }) => ({
    interval,
    percentage: parseFloat(percentageActive),
  }));

  const startCaseIntervalType = _startCase(intervalType);

  return (
    <IntervalGraph
      loading={loading}
      data={data}
      average={average(data)}
      title={`Percentage of Days per ${startCaseIntervalType} Active`}
      color="#9055A2"
      stroke="#2E294E"
      xAxisKey="interval"
      dataKey="percentage"
      metricName="% of Days with a Workout"
      unit={startCaseIntervalType}
      fixedTop={100}
    />
  );
}
