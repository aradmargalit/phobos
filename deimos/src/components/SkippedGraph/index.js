import SkippedGraph from './SkippedGraph';

export default SkippedGraph;
