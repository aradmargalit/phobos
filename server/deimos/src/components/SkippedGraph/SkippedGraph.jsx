import './SkippedGraph.scss';

import { Spin } from 'antd';
import { meanBy as _meanBy } from 'lodash';
import moment from 'moment';
import React from 'react';
import {
  Area,
  AreaChart,
  CartesianGrid,
  ReferenceLine,
  ResponsiveContainer,
  Tooltip,
  XAxis,
  YAxis,
} from 'recharts';

import AngledGraphTick from '../AngledGraphTick';

const transform = data =>
  data
    .sort((a, b) => moment(new Date(a.month)) - moment(new Date(b.month)))
    .map(({ month, days_skipped: daysSkipped }) => ({
      month,
      skipped: parseFloat(daysSkipped),
    }));

const calculateAverage = data => _meanBy(data, 'skipped');

export default function SkippedGraph({ loading, monthlyData }) {
  if (loading) return <Spin />;
  const data = transform(monthlyData);

  return (
    <div className="activity-graph-wrapper">
      <div className="graph-header">
        <h2>M O N T H L Y</h2>
        <h2>D A Y S</h2>
        <h2>S K I P P E D</h2>
      </div>
      <ResponsiveContainer width="100%" height={450}>
        <AreaChart
          className="activity-graph"
          data={data}
          margin={{ top: 10, right: 30, left: 0, bottom: 0 }}
          padding={{ top: 10, right: 30, left: 30, bottom: 10 }}
        >
          <defs>
            <linearGradient id="skipColor" x1="0" y1="0" x2="0" y2="1">
              <stop offset="5%" stopColor="#043927" stopOpacity={0.6} />
              <stop offset="95%" stopColor="#043927" stopOpacity={0} />
            </linearGradient>
          </defs>
          <XAxis
            interval={2}
            dataKey="month"
            height={120}
            tick={<AngledGraphTick />}
          />
          <YAxis />
          <CartesianGrid strokeDasharray="3 3" />
          <ReferenceLine
            y={calculateAverage(data)}
            stroke="#043927"
            strokeDasharray="3 3"
            label={{
              position: 'top',
              fontWeight: 600,
              value: 'Monthly Average',
            }}
          />
          <Tooltip
            separator={null}
            formatter={value => [`${value} days skipped`, '']}
            animationDuration={300}
          />
          <Area
            dataKey="skipped"
            type="monotone"
            stroke="#043927"
            fillOpacity={1}
            fill="url(#skipColor)"
          />
        </AreaChart>
      </ResponsiveContainer>
    </div>
  );
}
