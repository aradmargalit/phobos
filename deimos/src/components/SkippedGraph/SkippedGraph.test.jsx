import { shallow } from 'enzyme';
import React from 'react';

import SkippedGraph from './SkippedGraph';

const generateComponent = props => {
  const defaultProps = {};

  const mergedProps = { ...defaultProps, ...props };

  return shallow(<SkippedGraph {...mergedProps} />);
};

describe('<SkippedGraph />', () => {
  it('matches snapshots', () => {
    expect(
      generateComponent({
        monthlyData: [{ month: 'January 2020', days_skipped: 123 }],
      })
    ).toMatchSnapshot();
  });
});
