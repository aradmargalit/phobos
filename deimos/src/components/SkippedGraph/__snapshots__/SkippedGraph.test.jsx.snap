// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SkippedGraph /> matches snapshots 1`] = `ShallowWrapper {}`;
